extensions/awt/deployment/src/main/java/io/quarkus/awt/deployment/AwtProcessor.java

```diff
@@ -95,7 +98,11 @@ ReflectiveClassBuildItem setupReflectionClassesWithMethods() {
     void setupAWTInit(BuildProducer<JniRuntimeAccessBuildItem> jc,
             BuildProducer<JniRuntimeAccessMethodBuildItem> jm,
             BuildProducer<JniRuntimeAccessFieldBuildItem> jf,
-            NativeImageRunnerBuildItem nativeImageRunnerBuildItem) {
+            NativeImageRunnerBuildItem nativeImageRunnerBuildItem,
+            Optional<ProcessInheritIODisabled> processInheritIODisabled,
+            Optional<ProcessInheritIODisabledBuildItem> processInheritIODisabledBuildItem) {
+        nativeImageRunnerBuildItem.getBuildRunner()
+                .setup(processInheritIODisabled.isPresent() || processInheritIODisabledBuildItem.isPresent());
```

Comment:
If we have to do that, there is something wrong. I'll have to have a look later.

Reply:
Maybe I am misusing the `setup(...)` method, but I can say that `setup()` has not yet been called when `AwtProcessor` is executed. Without the call to `setup(...)`, the successive call to `nativeImageRunnerBuildItem.getBuildRunner().getGraalVMVersion()` would trigger the initial image pull.